src/main/java/org/apache/commons/io/IOUtils.java

```diff
@@ -2697,6 +2702,48 @@ public static byte[] toByteArray(final InputStream input, final long size) throw
         return toByteArray(input, (int) size);
     }
 
+    /**
+     * Reads exactly {@code size} bytes from the given {@link InputStream} into a new {@code byte[]}.
+     *
+     * <p>The method accumulates the data in temporary buffers of size at most {@code bufferSize}
```

Comment:
I think it would be nice to explain _why_ this method is useful, instead of _how_ it works. For example, "When reading from an untrusted input stream, you reduce the risk of an OOME by calling this API with...".